app/src/main/java/woowacourse/shopping/ui/detail/DetailViewModel.kt

```
@@ -0,0 +1,116 @@
+                val quantity = cartRepository.getCartItemQuantity(id)
+
+                recentItemRepository.addRecentItem(product)
+
+                _uiState.value =
+                    _uiState.value.copy(
+                        product = product.toUiModel(),
+                        quantity = quantity,
+                        recentItem = lastViewedItem?.toUiModel(),
+                        totalPrice = product.getPrice() * quantity,
+                    )
+            } catch (e: IllegalArgumentException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)